modules/core/src/main/java/com/google/refine/importing/ImportingUtilities.java

```diff
@@ -700,10 +699,11 @@ static public boolean postProcessRetrievedFile(
 
     static public void postProcessSingleRetrievedFile(File file, ObjectNode fileRecord) {
         if (!fileRecord.has("format")) {
-            JSONUtilities.safePut(fileRecord, "format",
-                    ImportingManager.getFormat(
-                            file.getName(),
-                            JSONUtilities.getString(fileRecord, "declaredMimeType", null)));
+            String encoding = JSONUtilities.getString(fileRecord, "declaredEncoding", null);
+            String bestFormat = ImportingManager.getFormat(file.getName(), JSONUtilities.getString(fileRecord, "declaredMimeType", null));
+            bestFormat = bestFormat == null ? "text" : bestFormat;
+            String format = guessBetterFormat(file, encoding, bestFormat);
+            JSONUtilities.safePut(fileRecord, "format", format);
```

Comment:
I would rather move this logic to a `FormatGuesser` which would be called consistently alongside other format-guessing heuristics.

Reply:
@wetneb Are you suggesting moving the logic to an overloaded guessBetterFormat method?

Reply:
Yes, that would be intuitively cleaner to me

Reply:
Code updated; please review

Reply:
Sorry, it looks like there was a misunderstanding again. I wrote above "Ideally in a new FormatGuesser implementation, to keep it separate from the existing ones which try to guess types via other means."

By this I expected that you would create a new class (say `com.google.refino.importers.ExtensionFormatGuesser`), in a new file. This class would `implement FormatGuesser`, and overload the only method offered by this interface: `public String guess(File file, String encoding, String seedFormat);`

It could look like this:
```java

public class ExtensionFormatGuesser implements FormatGuesser {
     public String guess(File file, String encoding, String seedFormat) {
          // ...
    }
}
```

Reply:
@wetneb FormatGuessers are defined for a format class, each formatter attenpt to guess a more correct format by sampling data. In the issue being addressed the file format is not available as the the client is not passing this info, the proposed solution addresses this by identifying the format from the pre-defined dataset of file extension to format, so it is not tied to a file format, which is why adding this in a FormatGuesser did not seem the right approach. 

Adding a format guesser for this would be something like this; does this seem right?

`public class ExtensionFormatGuesser implements FormatGuesser {

    @Override
    public String guess(File file, String encoding, String seedFormat) {
        String betterFormat = ImportingManager.getFormat(file.getName(), null);
        return betterFormat != null ? betterFormat : seedFormat ;
    }
}`

`    static String guessBetterFprmat(File file, ObjectNode fileRecord) {
        String encoding = JSONUtilities.getString(fileRecord, "declaredEncoding", null);
        FormatGuesser guesser = new ExtensionFormatGuesser();
        String betterFormat = guesser.guess(file, encoding, "text");
        return guessBetterFormat(file, encoding, betterFormat);
    }`

Reply:
@wetneb Can you review my previous note and let know next steps.